### DISCUSSION TITLE
clear cache after reseting the query variables.

### DISCUSSION THREAD

**Discussion by mahdi-jafaree** [urql-graphql/urql#3413] [Q&A]
Hi 
I'm using urql with react native in order to create a infinite scroll list. I went through the documentation and since my graphql api is relay compatible pagination, I used `relayPagination` exported from `@urql/exchange-graphcache/extras`. for caching as it's expected I'm using `graphcache` exchange.  
I want to add the pull-down-to-refresh when user tries to refresh the list item should only show the new data and clear the old data (same experience as the first query runs) but now when I call the `executeQuery` with varaibles being reset callback function returned by `useQuery` the old data is not cleared. also I want to keep old data while scrolling down. 
is it possible to have only new data when variables are reset? if yes may I know how? 
Thanks!

**Comment by wcastand** [comment]
Hi, did you ever found a solution to this ? currently having the same need in my expo app :/
